java/src/main/java/ai/rapids/cudf/Aggregation.java

```diff
@@ -0,0 +1,482 @@
+        private final boolean includeNulls;
+
+        public CountLikeAggregation(Kind kind, boolean includeNulls) {
+            super(kind);
+            this.includeNulls = includeNulls;
+        }
+
+        @Override
+        long createNativeInstance() {
+            return Aggregation.createCountLikeAgg(kind.nativeId, includeNulls);
+        }
+
+        @Override
+        public int hashCode() {
+            return 31 * kind.hashCode() + Boolean.hashCode(includeNulls);
+        }
+
+        @Override
+        public boolean equals(Object other) {
+            if (this == other) {
+                return true;
+            } else if (other instanceof CountLikeAggregation) {
+                CountLikeAggregation o = (CountLikeAggregation) other;
+                return o.includeNulls == this.includeNulls;
+            }
+            return false;
+        }
+    }
+
+    private static class QuantileAggregation extends Aggregation {
+        private final QuantileMethod method;
+        private final double[] quantiles;
+
+        public QuantileAggregation(QuantileMethod method, double[] quantiles) {
+            super(Kind.QUANTILE);
+            this.method = method;
+            this.quantiles = quantiles;
+        }
+
+        @Override
+        long createNativeInstance() {
+            return Aggregation.createQuantAgg(method.nativeId, quantiles);
+        }
+
+        @Override
+        public int hashCode() {
+            return 31 * (31 * kind.hashCode() + method.hashCode()) + Arrays.hashCode(quantiles);
+        }
+
+        @Override
+        public boolean equals(Object other) {
+            if (this == other) {
+                return true;
+            } else if (other instanceof QuantileAggregation) {
+                QuantileAggregation o = (QuantileAggregation) other;
+                return this.method == o.method && Arrays.equals(this.quantiles, o.quantiles);
+            }
+            return false;
+        }
+    }
+
+    protected final Kind kind;
+
+    protected Aggregation(Kind kind) {
+        this.kind = kind;
+    }
+
+    /**
+     * Add a column to the Aggregation so it can be used on a specific column of data.
+     * @param columnIndex the index of the column to operate on.
+     */
+    public AggregationOnColumn onColumn(int columnIndex) {
+        return new AggregationOnColumn(this, columnIndex);
+    }
+
+    /**
+     * <code>cudf::aggregation</code>, which is the equivalent of this class, is always used as
+     * a <code>std::unique_ptr&gt;cudf::aggregation&lt;</code>.  This means that ownership of the
+     * object is transferred when used.  We also don't want to force users to close an Aggregation.
+     * Both of these things together lead to a design where Aggregation objects are created in pure
+     * java, but when it is time to use them this method is called to return a pointer to a
+     * <code>std::unique_ptr</code> to the c++ aggregation instance. All values returned by this
```

Comment:
It looks like this comment block needs a slight update now that we're not returning a pointer to a `std::unique_ptr` anymore at the JNI layer.

Reply:
Great catch.  I have updated it.